<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Promise</title>
</head>
<body>
    <label for="zip">Zip Code:</label>
    <input type="number" id="zip" />  <!-- 創建可以給使用者輸入的地方, type 是輸入類型-->
    <button id="btnGetInfo">Get Info</button>
    <div id="info"></div>

    <script>
        const btnGetInfo = document.getElementById('btnGetInfo');

        btnGetInfo.onclick = () => getResponse();
        // await: 在 Promise 結束前, 後面的 Code 都無法被執行.
        // await 的錯誤會讓 async 拋出 error, 在正確執行下, 會 return resolved, 否則 return rejected 的狀態
        async function getResponse() {
            const zip = document.getElementById('zip').value;
            const zipURL = `http://api.zippopotam.us/us/${zip}`;
            const response = await fetch(zipURL);  
            const data = await response.json(); 
            console.log(data);
            displayResult(data);
        }

        displayResult = (data) => {
            const city = data.places[0]["place name"];
            const state = data.places[0].state;
            document.getElementById("info").innerHTML = `<h1>${city}, ${state}</h1>`;
        }
    </script>
</body>
</html>